packages/react-native-sdk/src/components/ConnectModalContent.tsx

```diff
@@ -0,0 +1,171 @@
+import { useState, useCallback } from "react";
+import { View, Image, StyleSheet, ActivityIndicator } from "react-native";
+import type { EmbeddedProviderAuthType } from "@phantom/embedded-provider-core";
+import { Button, Icon, Text, useTheme } from "@phantom/wallet-sdk-ui";
+import { usePhantom } from "../PhantomContext";
+import { useConnect } from "../hooks/useConnect";
+
+export interface ConnectModalContentProps {
+  appIcon?: string;
+  appName?: string;
+  onClose: () => void;
+}
+
+export function ConnectModalContent({ appIcon, onClose }: ConnectModalContentProps) {
+  const theme = useTheme();
+  const { isConnecting: contextIsConnecting, allowedProviders } = usePhantom();
+  const { connect } = useConnect();
+
+  const [isConnecting, setIsConnecting] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+  const [providerType, setProviderType] = useState<EmbeddedProviderAuthType | null>(null);
+
+  const isLoading = contextIsConnecting || isConnecting;
+
+  const errorBackgroundColor = `${theme.error}1A`;
+  const errorBorderColor = `${theme.error}4D`;
+  const errorTextColor = theme.error;
+
+  const connectWithAuthProvider = useCallback(
+    async (provider: EmbeddedProviderAuthType) => {
+      try {
+        setIsConnecting(true);
+        setError(null);
+        setProviderType(provider);
+
+        await connect({ provider });
+
+        onClose();
+      } catch (err) {
+        const error = err instanceof Error ? err : new Error(String(err));
+        setError(error);
+      } finally {
+        setIsConnecting(false);
+        setProviderType(null);
+      }
+    },
+    [connect, onClose],
+  );
+
+  const styles = StyleSheet.create({
+    appIcon: {
+      borderRadius: 28,
+      height: 56,
+      width: 56,
+    },
+    buttonContainer: {
+      alignItems: "center",
+      flexDirection: "column",
+      gap: 12,
+      width: "100%",
+    },
+    container: {
+      alignItems: "center",
+      flexDirection: "column",
+      gap: 24,
+      width: "100%",
+    },
+    errorContainer: {
+      backgroundColor: errorBackgroundColor,
+      borderColor: errorBorderColor,
+      borderRadius: parseInt(theme.borderRadius),
+      borderWidth: 1,
+      padding: 12,
+      width: "100%",
+    },
+    errorText: {
+      color: errorTextColor,
+      fontSize: 14,
+    },
+    loadingContainer: {
+      alignItems: "center",
+      flexDirection: "column",
+      gap: 12,
+      justifyContent: "center",
+      padding: 24,
+    },
+    spacer: {
+      flex: 1,
+    },
+  });
```

Comment:
This will be an object re-created on every re-render.
It should at least be wrapped in a `useMemo` and better it should be specified outside of the component and only the properties that can change with theme should be passed into components separately.